buffer.go

```diff
@@ -45,11 +51,26 @@ func (buffer *BaseBuffer) String() string {
 }
 
 func (buffer *BaseBuffer) Write(bytes []byte) (int, error) {
-	// TODO: only make new lines when we see a \n
-	buffer.lines = append(buffer.lines, string(bytes))
+	for _, rawLine := range strings.SplitAfter(string(bytes), "\n") {
```

Comment:
The write function no longer blindly appends each write call as a new line in the file. now it creates new lines only when it sees \n. This is kind of awesome because now we can use standard modules to fill up our buffers. 

For example:
to load a file into a buffer we could....

```go
f := os.Open("./main.go")
b := &BaseBuffer{}
io.Copy(b, f)
```

so cool!